pdr_backend/lake/etl.py

```diff
@@ -121,13 +129,80 @@ def do_bronze_step(self):
         end_ts = time.time_ns() / 1e9
         print(f"do_bronze_step - Completed in {end_ts - st_ts} sec.")
 
+    def _get_max_timestamp_values_from(
+        self, table_names: List, build_mode: bool = False
+    ) -> Dict:
+        """
+        @description
+            Get the max timestamp values from the tables
+
+        @arguments
+            table_names - The list of table names to get the max timestamp values from
+
+        @returns
+            The max timestamp values from the tables
+        """
+
+        pds = PersistentDataStore(self.ppss.lake_ss.parquet_dir)
+        max_timestamp_query = "SELECT MAX(timestamp) as max_timestamp FROM {}"
+        values = {}
+        for table_name in table_names:
+            calc_table_name = get_table_name(table_name, build_mode)
+            result = pds.query_data(max_timestamp_query.format(calc_table_name))
+            values[calc_table_name] = (
+                result["max_timestamp"][0] if result is not None else None
+            )
+
+        # print(f"1111_get_max_timestamp_values_from - {values}")
+        return values
+
+    def _calc_bronze_start_end_ts(self):
+        """
+        @description
+            Calculate the start and end timestamps for the bronze tables
+            ETL updates should use from_timestamp by calculating
+            max(etl_tables_max_timestamp).
+            ETL updates should use to_timestamp by calculating
+            min(max(source_tables_max_timestamp)).
+        """
+        from_values = self._get_max_timestamp_values_from(
+            self.bronze_table_names
+        ).values()
+        from_timestamp = (
+            min(from_values)
+            if None not in from_values
+            else datetime.strptime(self.ppss.lake_ss.st_timestr, "%Y-%m-%d_%H:%M")
+        )
+
+        to_values = self._get_max_timestamp_values_from(
+            self.raw_table_names, build_mode=True
+        ).values()
+        to_timestamp = (
+            min(to_values)
+            if None not in to_values
+            else datetime.strptime(self.ppss.lake_ss.fin_timestr, "%Y-%m-%d_%H:%M")
+        )
```

Comment:
This is what I meant by us being able to leverage SQL + Views (not tables) to simplify this.

pseudo code 1 - create view that unions both tables so all data sits together
views should not take more storage/mem, it's "an alias for two different tables"
```
create view etl_bronze_pdr_predictions (
select *
from bronze_pdr_predictions
union etl_bronze_pdr_predictions
);
```

pseudocode 2 - reduce query across tables to 1 scan/execute
```
"SELECT MAX(timestamp) as max_timestamp FROM etl_bronze_pdr_predictions"
```